docs/source/user_migration_guides.md

```diff
@@ -0,0 +1,92 @@
+Migrating user code
+===================
+
+These are migration guides aimed specifically at user of ipywidgets.
+
+Migrating from 7.x to 8.0
+-------------------------
+
+For more details about the changes done for the 8.0 major version, please consult the
+[changelog](./changelog).
+
+### Code
+
+#### FileUpload
+
+The `data` and `metadata` traits have been removed, and the `value` trait revamped to
+be a list of dicts containing the file information. The keys of these dicts are:
+
+- `content`: The buffer of file data
+- `name`: The name of the file
+- `type`: The MIME type of the file content
+- `size`: The size of the buffer in bytes
+- `last_modified`: A UTC datetime representing the "last modified" value reported for the file
+
+Suggested migration: Rewrite all usage of `FileUpload` to use the new structure.
+If you need to support both 7.x and 8.x, you can e.g. write functions `get_file_buffer` and similar
+to wrap reads from the widget.
```

Comment:
This seems to me to be the largest breaking change for users. I.e. it is likely to completely break code (exceptions will be raised that cannot be recovered from), and is likely to be heavily used.